### PR TITLE
Removing unused test code

### PR DESCRIPTION
**Pull request by john-science**
## Description

Based on discussion in [this ticket](https://github.com/terrapower/armi/issues/1133), I ran our unit tests with a special code coverage setting, to determine which of our unit tests weren't being run.

I have removed unused tests, or cleaned them up, as seemed appropriate.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by albeanth** [comment]
> I ran our unit tests with a special code coverage setting,

What is the special voodoo magic you are using?! Super cool.

**Comment by opotowsky** [comment]
> > I ran our unit tests with a special code coverage setting,
> 
> What is the special voodoo magic you are using?! Super cool.

This was my initial reaction too

File: armi/nuclearDataIO/tests/test_xsLibraries.py [hunk 1]
```diff
@@ -59,122 +54,6 @@
...
-def copyInputForPmatrxAndGamsio(inpPath):
-    with open(inpPath, "r") as inp, open(
-        inpPath.replace(".inp", ".pmatrx.inp"), "w"
-    ) as pmatrxInp, open(inpPath.replace(".inp", ".gamiso.inp"), "w") as gamisoInp:
-        for line in inp:
-            pmatrxInp.write(line.replace("isotxs", "pmatrx"))
-            gamisoInp.write(line.replace(".isotxs", ".gamiso"))
-
-
-def createTestXSLibraryFiles(cachePath):
```
Comment (opotowsky) [inline armi/nuclearDataIO/tests/test_xsLibraries.py:71]: Lazy question: The files mentioned in this test are still relevant to other tests correct?

**Review by opotowsky — APPROVED** [review]
I'm surprised to see the 2 LFP tests weren't being run at all.

**Comment by john-science** [comment]
> > > I ran our unit tests with a special code coverage setting,
> > What is the special voodoo magic you are using?! Super cool.
> This was my initial reaction too

Sorry, I was trying to not go too deep into the nitty-gritty.

It was easy. I went into our `.coveragerc` file and delete this line:

https://github.com/terrapower/armi/blob/6c397f66dd705b3baef3219658a9b1d480b40241/.coveragerc#L20

That's the line that means we explicitly IGNORE our unit tests when we calculate code coverage. This is important, otherwise we would get heavily inflated code coverage.

Anyway, I removed that one line, and ran HTML code coverage locally, then checked the results with my eye balls.

**Fun Fact**: One [my first PRs](https://github.com/terrapower/armi/pull/305) into ARMI (before I started working at TerraPower) was to add this line to the `.coveragerc`, because our code coverage was heavily inflated.

File: armi/nuclearDataIO/tests/test_xsLibraries.py (hunk 1 above)
Comment (john-science) [reply armi/nuclearDataIO/tests/test_xsLibraries.py:71]: That's a good question.  I'm like 95% sure, yes.

Comment (mgjarrett) [reply armi/nuclearDataIO/tests/test_xsLibraries.py:71]: This isn't a unit test, but a script that creates MC2-3 inputs to generate XS libraries to be used in unit tests. It probably doesn't belong in a unit test file, but it seems like we might still want to retain the functionality in case we want to update these files in the future? e.g., if we use a new version of MC2-3 that changes the format of the library files?

Comment (opotowsky) [reply armi/nuclearDataIO/tests/test_xsLibraries.py:71]: I really appreciate the context. Looks like we moved too fast. 

https://github.com/terrapower/armi/issues/1174

File: armi/physics/neutronics/fissionProductModel/tests/test_fissionProductModel.py [hunk 2]
```diff
@@ -75,32 +66,6 @@ def test_getAllFissionProductNames(self):
         self.assertIn("XE135", fissionProductNames)
 
 
-class TestFissionProductModelIndependentLumpedFissionProducts(unittest.TestCase):
```
Comment (jakehader) [inline armi/physics/neutronics/fissionProductModel/tests/test_fissionProductModel.py:78]: Wait, why is this removed now?

Comment (john-science) [reply armi/physics/neutronics/fissionProductModel/tests/test_fissionProductModel.py:78]: It was unused code.

It had a `setUp()` method, but no tests. So no tests were ever run. Even the `setUp()` method was never run, which I proved to myself by putting an `assert False` in it.

Comment (opotowsky) [reply armi/physics/neutronics/fissionProductModel/tests/test_fissionProductModel.py:78]: John ran code coverage on the unit tests to see which ones weren't running, and did some cleanup.

Comment (opotowsky) [reply armi/physics/neutronics/fissionProductModel/tests/test_fissionProductModel.py:78]: (jinx)
